File: Spring/les/src/main/java/com/jga/les/service/ProdutoService.java
package com.jga.les.service;

import com.jga.les.model.Produto;
import com.jga.les.repository.ProdutoRepository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;

@Service
public class ProdutoService extends GenericService<Produto, Long> {

    public ProdutoService(JpaRepository<Produto, Long> objRepository) {
        super(objRepository, Produto.class);
    }

    public Produto findByCodigo(String codigo) {
        return ((ProdutoRepository) objRepository).findByCodigo(codigo);
    }
}

File: Flutter/les/lib/services/produto_service.dart

import 'package:dio/dio.dart';
import 'package:les/core/endpoints.dart';
import 'package:les/core/generic_service.dart';
import 'package:les/model/produto/produto.dart';
import 'package:result_dart/result_dart.dart';

class ProdutoService extends GenericService<Produto> {

  ProdutoService(Dio dio) : super(dio, 'produto', (data) => Produto.fromJson(data));

  AsyncResult<String> imprimirCodigoBarras(String codigo, int qtde) async {
    try{
      await dio.post('${Endpoints.baseUrl}/printerdl200/barcode?data=$codigo&qntd=$qtde');
      return Success('Impresso com sucesso.');
    } catch (e) {
      return Failure(Exception(e.toString()));
    }
  }

}